trustmanager/keystore.go

```diff
@@ -265,49 +266,54 @@ func copyKeyInfoMap(keyInfoMap map[string]KeyInfo) map[string]KeyInfo {
 // KeyInfoFromPEM attempts to get a keyID and KeyInfo from the filename and PEM bytes of a key
 func KeyInfoFromPEM(pemBytes []byte, filename string) (string, KeyInfo, error) {
 	keyID, role, gun := inferKeyInfoFromKeyPath(filename)
-	if role == "" {
-		block, _ := pem.Decode(pemBytes)
-		if block == nil {
-			return "", KeyInfo{}, fmt.Errorf("could not decode PEM block for key %s", filename)
-		}
-		if keyRole, ok := block.Headers["role"]; ok {
-			role = keyRole
-		}
+	keyID = filepath.Base(filename)
+	block, _ := pem.Decode(pemBytes)
+	if block == nil {
+		return "", KeyInfo{}, fmt.Errorf("could not decode PEM block for key %s", filename)
+	}
+	if keyRole, ok := block.Headers["role"]; ok {
+		role = keyRole
+	}
+	if keyGun, ok := block.Headers["gun"]; ok {
+		gun = keyGun
 	}
 	return keyID, KeyInfo{Gun: gun, Role: role}, nil
 }
 
 // getKeyRole finds the role for the given keyID. It attempts to look
 // both in the newer format PEM headers, and also in the legacy filename
-// format. It returns: the role, whether it was found in the legacy format
-// (true == legacy), and an error
-func getKeyRole(s Storage, keyID string) (string, bool, error) {
+// format. It returns: the role, whether it was found in the legacy(0.1) format
+// (true == legacy),  whether it was found in a notary0.3 formal (true == notary0.3) and an error
+func getKeyRole(s Storage, keyID string) (string, bool, bool, error) {
```

Comment:
instead of returning two booleans for each legacy format (which could potentially grow), how about instead returning the path of the key relative to the base `private` directory as a string?  Then the `legacy` and `notary03` checks can go away, and all accesses to that path are handled generically?